RESOURCE_ACCESS_CONTROL_FOR_PLUGINS.md

```diff
@@ -0,0 +1,511 @@
+      }
+    }
+  }
+}
+```
+
+Note: `default` is a place-holder action-group that will eventually be replaced with actual action-groups in the future.
+
+#### **Response Fields:**
+| Field          | Type    | Description |
+|---------------|---------|-------------|
+| `sharing_info` | Object  | Contains information about how the resource is shared. |
+| `source_idx`   | String  | The OpenSearch index containing the resource. |
+| `resource_id`  | String  | The unique identifier of the resource being shared. |
+| `created_by`   | Object  | Information about the user who created the sharing entry. |
+| `share_with`   | Object  | Defines users, roles, and backend roles with access to the resource. |
+
+---
+
+### **3. Revoke Access**
+- **Endpoint:**
+  ```
+  POST /_plugins/_security/resources/revoke
+  ```
+- **Description:**
+  Revokes access to a resource for specific users, roles, or backend roles.
+
+#### **Request Body:**
+```json
+{
+  "resource_id": "my-resource",
+  "resource_index": "resource-index",
+  "entities_to_revoke": {
+    "roles": ["shared-roles"]
+  }
+}
+```
+
+#### **Request Fields:**
+| Field            | Type    | Description |
+|-----------------|---------|-------------|
+| `resource_id`   | String  | The unique identifier of the resource whose access is being revoked. |
+| `resource_index`| String  | The OpenSearch index where the resource is stored. |
+| `entities_to_revoke` | Object | Specifies which **users, roles, or backend roles** should have their access removed. |
+| `roles`        | Array   | List of roles to revoke access from. |
+
+#### **Response:**
+Returns the updated **resource sharing state** after revocation.
+```json
+{
+  "sharing_info": {
+    "source_idx": "resource-index",
+    "resource_id": "my-resource",
+    "created_by": {
+      "user": "admin"
+    },
+    "share_with": {
+      "default": {
+        "users": ["shared-user-name"],
+        "backend_roles": ["shared-backend-roles"]
+      }
+    }
+  }
+}
+```
+
+Note: `default` is a place-holder action-group that will eventually be replaced with actual action-groups in the future.
+
+#### **Response Fields:**
+| Field          | Type    | Description |
+|---------------|---------|-------------|
+| `sharing_info` | Object  | Contains information about the updated resource sharing state. |
+| `source_idx`   | String  | The OpenSearch index containing the resource. |
+| `resource_id`  | String  | The unique identifier of the resource. |
+| `created_by`   | Object  | Information about the user who created the sharing entry. |
+| `share_with`   | Object  | Defines users, roles, and backend roles that still have access to the resource. |
+
+---
+
+### **4. List Accessible Resources**
+- **Endpoint:**
+  ```
+  GET /_plugins/_security/resources/list/{resource_index}
+  ```
+- **Description:**
+  Retrieves a list of **resources that the current user has access to** within the specified `{resource_index}`.
+
+#### **Response:**
```

Comment:
What are the performance implications of the "extra hop" with the resource sharing security index?

Reply:
As we discussed during the design phase, there are is no concept of joins in OpenSearch. So the alternative to current approach would be to store resource data inside resource-sharing index, which results in significant amount of duplication and something we are moving away from.